### PR TITLE
Publish all installers/archives for sharedfx, host, and the sdk

### LINKED ISSUE #2007: Ubuntu Latest links are not getting updated with recent builds
The installer links on the [Readme](https://github.com/dotnet/cli/blob/rel/1.0.0/README.md) for Ubuntu are out of date.  The readme indicates the current version is 1.0.0.001971 yet the installers are for 1.0.0.001793.

https://dotnetcli.blob.core.windows.net/dotnet/beta/Installers/Latest/dotnet-ubuntu-x64.latest.deb
https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-dev-ubuntu-x64.latest.tar.gz

I took a cursory look and it appears this regression may have been introduced during the refactoring work to replace the scripts with build targets.  I do not see any logic that is updating the latest installer links for ubuntu builds - https://github.com/dotnet/cli/blob/rel/1.0.0/scripts/dotnet-cli-build/PublishTargets.cs

### PR DESCRIPTION
**Pull request by brthor**
Publishing for debs, pkgs, msis, with appropriate versions dependent on the package type (host, sdk, sharedfx)

Also unify versions of all installers to the semantic version.

Fixes #2007 

Worked with @krwq to get this tested and ready.

cc @Sridhar-MS @piotrpMSFT

### CODE REVIEW

**Comment by brthor** [comment]
@dotnet-bot test Windows_2016 x86 Debug Build please

File: packaging/host/debian/dotnet-sharedhost-debian_config.json [hunk 1]
```diff
@@ -28,6 +28,11 @@
         "full_text": "Copyright (c) 2015 Microsoft\nPermission is hereby granted, free of charge, to any person obtaining a copy\nof this software and associated documentation files (the \"Software\"), to deal\nin the Software without restriction, including without limitation the rights\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\ncopies of the Software, and to permit persons to whom the Software is\nfurnished to do so, subject to the following conditions:\nThe above copyright notice and this permission notice shall be included in all\ncopies or substantial portions of the Software.\nTHE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\nSOFTWARE."
     },
 
+    "package_conflicts" : [
+        "dotnet",
```
Comment (ellismg) [inline packaging/host/debian/dotnet-sharedhost-debian_config.json]: Ultimately we are going to want `dotnet` to be a package that just has dependencies on the shared host package and a shared framework (of some version), correct?  At that point, do we remove this?

Comment (brthor) [inline packaging/host/debian/dotnet-sharedhost-debian_config.json]: The concern here is that the current `dotnet` package will conflict if anyone were to try to install the new packages alongside. Having it here until we publish the `dotnet` package makes that a little smoother I think.

**Comment by piotrpMSFT** [comment]
What's the `dotnet` package we're referring to here? `dotnet-1.0.0`?

File: scripts/dotnet-cli-build/PublishTargets.cs [hunk 2]
```diff
@@ -12,32 +12,35 @@ namespace Microsoft.DotNet.Cli.Build
...
+            SharedFrameworkNugetVersion = c.BuildContext.Get<string>("SharedFrameworkNugetVersion");
             Channel = c.BuildContext.Get<string>("Channel");
+
             return c.Success();
         }
 
         [Target(nameof(PrepareTargets.Init),
         nameof(PublishTargets.InitPublish),
-        nameof(PublishTargets.PublishArtifacts),
-        nameof(PublishTargets.TriggerDockerHubBuilds))]
```
Comment (MichaelSimons) [inline scripts/dotnet-cli-build/PublishTargets.cs]: Why was this removed?  I'm not finding it anywhere else?  Docker builds are no longer being triggered.

Comment (brthor) [inline scripts/dotnet-cli-build/PublishTargets.cs]: I meant to move it in the refactoring to happen after the deb files were published. Mixup, will fix ASAP
